src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py

```diff
@@ -927,6 +927,391 @@ def test_appconfig_to_appconfig_import_export(self, resource_group, location):
+        })
+        self.cmd('appconfig kv set --connection-string {src_connection_string} --key {key} --value {value} --content-type {content_type} -y',
+                 checks=[self.check('key', entry_key),
+                         self.check('value', appconfig_value),
+                         self.check('contentType', json_content_type_09)])
+
+        entry_key = "Key10"
+        entry_value = '[\\"text\\",true,null]'
+        appconfig_value = entry_value.replace('\\', '')
+        json_content_type_10 = 'application/json+hybridarray'
+        self.kwargs.update({
+            'key': entry_key,
+            'value': entry_value,
+            'content_type': json_content_type_10
+        })
+        self.cmd('appconfig kv set --connection-string {src_connection_string} --key {key} --value {value} --content-type {content_type} -y',
+                 checks=[self.check('key', entry_key),
+                         self.check('value', appconfig_value),
+                         self.check('contentType', json_content_type_10)])
+
+        # Validate that input value is in JSON format
+        entry_key = "Key11"
+        entry_value = 'Red'
+        self.kwargs.update({
+            'key': entry_key,
+            'value': entry_value,
+            'content_type': json_content_type_01
+        })
+        with self.assertRaisesRegexp(CLIError, "is not a valid JSON object, which conflicts with the content type."):
+            self.cmd('appconfig kv set --connection-string {src_connection_string} --key {key} --value {value} --content-type {content_type} -y')
+
+        self.kwargs.update({
+            'value': '[abc,def]'
+        })
+        with self.assertRaisesRegexp(CLIError, "is not a valid JSON object, which conflicts with the content type."):
+            self.cmd('appconfig kv set --connection-string {src_connection_string} --key {key} --value {value} --content-type {content_type} -y')
+
+        self.kwargs.update({
+            'value': 'True'
+        })
+        with self.assertRaisesRegexp(CLIError, "is not a valid JSON object, which conflicts with the content type."):
+            self.cmd('appconfig kv set --connection-string {src_connection_string} --key {key} --value {value} --content-type {content_type} -y')
+
+        # Create a non-JSON key-value and update its content type in subsequent command
+        self.kwargs.update({
+            'value': entry_value
+        })
+        self.cmd('appconfig kv set --connection-string {src_connection_string} --key {key} --value {value} -y',
+                 checks=[self.check('key', entry_key),
+                         self.check('value', entry_value)])
+
+        with self.assertRaisesRegexp(CLIError, "Set the value again in valid JSON format."):
+            self.cmd('appconfig kv set --connection-string {src_connection_string} --key {key} --content-type {content_type} -y')
+
+        # Add a new feature flag
+        entry_feature = 'Beta'
```

Comment:
Why we need to test feature flag and key vault reference in this testcase?

Reply:
To make sure that we can import/export them between config stores in the next test. I should move it to test scenario 2 section.

---
In reply to: [451843548](https://github.com/Azure/azure-cli/pull/14170#discussion_r451843548) [](ancestors = 451843548)